response_operations_ui/controllers/notify_controller.py

```diff
@@ -39,16 +39,16 @@ def _send_message(self, email, template_id, personalisation=None, reference=None
             notification.update({"reference": reference})
 
         url = f'{self.notify_url}{template_id}'
-
         auth = app.config['SECURITY_USER_NAME'], app.config['SECURITY_USER_PASSWORD']
         response = requests.post(url, auth=auth, json=notification)
-        if response.status_code == 201:
+        try:
+            response.raise_for_status()
             logger.info('Notification id sent via Notify-Gateway to GOV.UK Notify.', id=response.json().get("id"))
-        else:
+        except HTTPError:
             ref = reference if reference else 'reference_unknown'
-            raise NotifyError("There was a problem sending a notification to Notify-Gateway to GOV.UK Notify",
-                              reference=ref)
-
+            logger.error(error=response.text, status_code=response.status_code, ref=ref)
+            raise NotifyError('There was a problem sending a notification to Notify-Gateway')
```

Comment:
I should've just said to change it to this `raise NotifyError('There was a problem sending a notification to Notify-Gateway', error=response.text)` instead of explaining my thought process >_<

If we have the useful text that we can put into the exception, then we should add it as it might be needed in the future and would be frustrating to find out that the NotifyError offers the error text but nothing fills it in!